### PR TITLE
[LLM Runtime] dynamic link the layer to compress binary size

### PR DESCRIPTION
**Pull request by a32543254**
## Type of Change

feature or bug fix or documentation or others
API changed or not

## Description

detail description 
JIRA ticket: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/ITREX-cpp-graph-extension/232/

**Review by hshen14 — APPROVED** [review]

**Review by kevinintel — APPROVED** [review]

File: intel_extension_for_transformers/llm/runtime/graph/core/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ find_package(Threads REQUIRED)
 file(GLOB layers_srcs "layers/*.cpp")
 set(sources ne_layers.c ${layers_srcs})
 
-add_library_w_warning(ne_layers "${sources}")
+add_library_w_warning(ne_layers SHARED "${sources}")
```
Comment (luoyu-intel) [inline intel_extension_for_transformers/llm/runtime/graph/core/CMakeLists.txt:19]: hard-code change will not generate any exported symbols on Windows. It's not related to BUILD_SHARED_LIBS

**Review by luoyu-intel — CHANGES_REQUESTED** [review]

File: intel_extension_for_transformers/llm/runtime/graph/core/CMakeLists.txt (hunk 1 above)
Comment (DDEle) [reply intel_extension_for_transformers/llm/runtime/graph/core/CMakeLists.txt]: Thanks! 3918179 should work now.

**Review by airMeng — APPROVED** [review]

**Comment by VincyZhang** [comment]
final test: https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/ITREX-cpp-graph-extension/234/

File: intel_extension_for_transformers/llm/runtime/graph/cmake/Common.cmake [hunk 2]
```diff
@@ -36,9 +36,28 @@ function(add_executable_w_warning TARGET)
...
+    add_library_w_warning_(${TARGET} STATIC ${ARGN})
+endfunction()
+
+function(add_shared_library_w_warning TARGET)
+    add_library_w_warning_(${TARGET} SHARED ${ARGN})
+endfunction()
+
+function(add_shareable_library_w_warning TARGET)
+    if (BUILD_SHARED_LIBS)
+    add_library_w_warning_(${TARGET} SHARED ${ARGN})
```
Comment (luoyu-intel) [inline intel_extension_for_transformers/llm/runtime/graph/cmake/Common.cmake:56]: format spaces?

**Review by luoyu-intel — APPROVED** [review]

File: intel_extension_for_transformers/llm/runtime/graph/cmake/Common.cmake (hunk 2 above)
Comment (DDEle) [reply intel_extension_for_transformers/llm/runtime/graph/cmake/Common.cmake:56]: Ther is no formal format for cmake? Can be changed in next PR

**Pull request merged**